apps/api/src/campaign/campaign.service.ts

```diff
@@ -573,26 +554,88 @@ export class CampaignService {
         and status: ${newDonationStatus} because the event comes after existing donation with status: ${donation.status}`,
       )
     }
+  }
 
-    //For successful donations we will also need to link them to user and add donation wish:
-    if (newDonationStatus === DonationStatus.succeeded) {
-      Logger.debug('metadata?.isAnonymous = ' + metadata?.isAnonymous)
+  private async createIncomingDonation(
+    tx: Prisma.TransactionClient,
+    paymentData: PaymentData,
+    newDonationStatus: DonationStatus,
+    campaign: Campaign,
+  ) {
+    Logger.debug(
+      'No donation exists with extPaymentIntentId: ' +
+        paymentData.paymentIntentId +
+        ' Creating new donation with status: ' +
+        newDonationStatus,
+    )
 
-      if (metadata?.isAnonymous != 'true') {
-        await this.prisma.donation.update({
-          where: { id: donation.id },
-          data: {
-            person: {
-              connect: {
-                email: paymentData.billingEmail,
-              },
-            },
-          },
-        })
+    /**
+     * Create or connect campaign vault
+     */
+    const vault = await this.prisma.vault.findFirst({ where: { campaignId: campaign.id } })
```

Comment:
shouldn't we be using the `tx` object, just to be sure that we are always seeing the state of each record as it is in the transaction and prevent any inconsistencies

Reply:
yes, thanks! 
this is important as the donation creation could fail, if another donation creates the same vault in meantime